modules/Database/Arangojs.js

```diff
@@ -863,6 +863,27 @@ class ArangoJS {
         return count;
     }
 
+    /**
+     * This method will leave only encryption for selected color, datasetId and offerId
+     * Rest of the encryption data will be removed
+     * @param datasetId
+     * @param offerId
+     * @param leaveColor
+     * @returns {Promise<void>}
+     */
+    async removeUnnecessaryEncryptionData(datasetId, offerId, leaveColor) {
+        const queryString = `LET datasetMetadata = DOCUMENT('ot_datasets', @datasetId)
+for v in DOCUMENT('ot_vertices', datasetMetadata.vertices)
+filter v.encrypted != null
+filter v.encrypted[@offerId] != null
+let newEncrypted = merge(v.encrypted, {@offerId: { @leaveColor: v.encrypted[@offerId][@leaveColor]}})
+
+UPDATE v with {encrypted: newEncrypted } in 'ot_vertices' OPTIONS { mergeObjects: false }`;
+        await this.runQuery(queryString, {
+            datasetId, offerId, leaveColor,
+        });
+    }
```

Comment:
This function is expected to go through most of the vertices in a database, we should probably test this on a sufficiently filled server before committing to this migration.